subprojects/resources-http/src/main/java/org/gradle/internal/resource/transport/http/JavaSystemPropertiesProxySettings.java

```diff
@@ -58,11 +58,12 @@ private int initProxyPort(String proxyPortString) {
             return defaultPort;
         }
         try {
+            proxyPortString = proxyPortString.trim();
```

Comment:
How does this interact with the isBlank call above?

Reply:
And should we be trimming everything that we read from system properties?

Reply:
`isBlank` checks that `proxyPortString` is not null or blank (just empty characters). So at least a null check would still be needed here if we remove it.

> And should we be trimming everything that we read from system properties?
I guess we could trim all system properties in this class. I am not sure just for username and password, does it make sense to have empty characters there?

Reply:
@big-guy I changed the logic so we now trim all system properties. Could you take another look?